api/src/main/java/org/openmrs/module/appointments/notification/SmsSender.java

```diff
@@ -0,0 +1,7 @@
+package org.openmrs.module.appointments.notification;
+
+import java.util.Date;
+
+public interface SmsSender {
+    void send(String recipientMobileNumber, String patientName, String smsType, String serviceName, Date startDateTime, String teleLink);
```

Comment:
I was hoping the interface is Channel vocabulary based and not Appointment vocabulary based. Something like:

MessageSender.send(String phoneNumber, String fullMessage).

Then SMS Sender, Whatsapp Sender, etc can easily implement this interface. They don't care if the message is goign to patient, doctor, customer, relative, etc. They just send a message over the appropriate channel to the number. Its the caller of these classes that constructs the message and delegates the "sending" to the channel implementation.

So.. Appointment ---delegates to ---> AppointmentMessage class ---delegates to--> SMSSender class ---> sends SMS

Similarly,
Appointment ---delegates to ---> AppointmentMessage class ---delegates to (based on config)--> WhatsappSender class ---> sends SMS

Then we write unit tests for AppointmentMessage class which checks if the construction of message is correct based on passed parameters like PatientName, Service Name, Date, Link.

With this logic, the MessageSender interface and its implementations are reusable in the context of different messages (Appointments, Registration, Teleconsult, Doctor-Doctor consult, etc). 

For this JIRA ticket, we would only implement SMS (and not whatzapp classes).

Reply:
Building of the message text requires sms template identifier. The message template format is available in SMSSender class (DefaultSmsSender) which is needed as outgoing SMS has to be conforming to this format.
Either message template is available in config file or in global properties in openmrs.
Please advise